### PR TITLE
Preserve Whitespace in string Results

### LINKED ISSUE #314: Spaces are missing in output strings
If I have a string with any leading spaces, or multiple consecutive spaces, they are removed when showing the output. For example, this:
```clojure
;; A string with spaces:
"    hello     there     "
```
It is rendered as:
image
The leading spaces are removed, and all groups of more than one space are reduced to a single space.

I'm not sure if this is related to #306, but it feels different, so I opened a separate issue.

### PR DESCRIPTION
**Pull request by zampino**
Fixes #314.

### CODE REVIEW

File: src/nextjournal/clerk/render.cljs [hunk 1]
```diff
@@ -414,16 +414,12 @@
           (if (string? s) [s] s))))
 
 (defn render-quoted-string [s {:as opts :keys [path viewer !expanded-at] :or {path []}}]
```
Comment (zampino) [inline src/nextjournal/clerk/render.cljs:416]: Took the opportunity to simplify this, assuming presented nodes with string values will always give just a string value. Is this correct?

**Comment by emlyn** [comment]
I just tried this, and it looks much better, but there is still an extra newline inserted at the start of the string (if you copy the rendered result and paste it somewhere), although it is not displayed in the browser.

**Comment by zampino** [comment]
@emlyn thanks for testing this stuff :-) 

> there is still an extra newline inserted at the start of the string (if you copy the rendered result and paste it somewhere), although it is not displayed in the browser.

I'm afraid that's due to some markup that we need in order to display un/collapsed strings correctly. Especially when they're part of nested collections. But we'll investigate. 

As a partial workaround: if your selection doesn't span the surrounding quotes then the string should be copied as is.

**Comment by emlyn** [comment]
The newline thing is really nitpicking, this is great. Thanks for the fast fixes!

**Comment by mk** [comment]
@emlyn I've merged the PR as-is, think we'll handle copying of results by some other means in the future.

**Pull request merged**